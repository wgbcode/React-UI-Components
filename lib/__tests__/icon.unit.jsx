import * as renderer from "react-test-renderer";
import Icon from "../icon/icon";
import * as React from "react";
import { mount } from "enzyme";

describe("Icon 组件", () => {
  it("渲染成功", () => {
    // 将 React 组件渲染成 JS 对象，再变成字符串。
    const json = renderer.create(<Icon name="testIcon" />).toJSON();
    // 断言 json 字符串和 snap 文件中的字符串一致
    expect(json).toMatchSnapshot();
  });
  it("onClick", () => {
    const fn = jest.fn();
    const component = mount(<Icon name="testIcon" onClick={fn} />);
    component.find("svg").simulate("click");
    expect(fn).toBeCalled();
  });
});
