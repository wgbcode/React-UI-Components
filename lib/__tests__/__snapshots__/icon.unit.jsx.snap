// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon 组件 渲染成功 1`] = `
<span>
  <svg
    className="fui-icon"
  >
    <use
      xlinkHref="#testIcon"
    />
  </svg>
</span>
`;
